### ISSUE TITLE
SpecWithJUnit ignores some failed tests when running in maven

### ISSUE THREAD

**Issue by laurynasl-wix** [etorreborre/specs2#672]
Hi, 

There is an issue with a case like this:

```scala
import org.specs2.mutable.SpecWithJUnit
import org.specs2.specification.Scope

class MissingFailureTest extends SpecWithJUnit {
  
  "should fail" in new Context {
    1 must be_===(2)
  }
  
  trait Context extends Scope {
    throw new Error("Should fail")
  }
}
```

This is a reduced example from a real scenario when using JMock (it's UnexpectedInvocation exception extends error). It seems that surefire test runner does not get notified that a test failed. I think the Error gets dropped in `ActionRunOps.runOption`

This is a runnable maven example project https://github.com/laurynasl-wix/specs2-junit-issue/blob/master/src/test/scala/example/MissingFailureTest.scala

**Comment by etorreborre** [comment]
I have attempted a fix. Any mock framework or testing library integrated with specs2 needs to be specially handled if it uses `java.lang.Error` to model failures. JMock's class is `ExpectationError` and I tried to deal with it the same way I do with `AssertionError` and JUnit. I published `4.3.0-9613e1025-20180617171339` but my local testing  is currently not working, can you please try on your side?

**Comment by laurynasl-wix** [comment]
I've tested it in our setup and the test fails as expected. Thank you!
